plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcRecordCursor.java

```diff
@@ -135,8 +143,19 @@ public boolean advanceNextPosition()
 
         try {
             if (resultSet == null) {
-                log.debug("Executing: %s", statement.toString());
-                resultSet = statement.executeQuery();
+                try {
+                    resultSet = resultSetFuture.get();
+                }
+                catch (ExecutionException e) {
+                    if (e.getCause() instanceof SQLException) {
+                        throw (SQLException) e.getCause();
```

Comment:
This will have an "async stacktrace", so it's not obviously correct thing to do.
I saw once you had some more elaborate repackaing.

If not, add some explanatory comment about why doing so